compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -61,7 +61,7 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
       longName = base+"$"+suffix
     }
     usedLocalNames += longName
-    longName
+    mangleJSName(longName)
```

Comment:
Ah OK. I guess in that case changing to `isReserved` is probably the right thing to do.